### DISCUSSION TITLE
Old SF Cluster Source Code

### DISCUSSION THREAD

**Discussion by BryceIsaacTiglon** [official-stockfish/Stockfish#5388] [General]
I see that there is source code for the cluster version of the most recent SF 16.1. Is there somewhere where I can find source code for earlier cluster versions? I can find the standard versions of past SF models, but not the cluster versions.

**Comment by vondele** [comment]
All versions of the SF cluster code can be found in the corresponding cluster branch on github see https://github.com/official-stockfish/Stockfish/commits/cluster/

**Comment by vondele** [comment]
The actual versions of the cluster branch correspond to merge commits as in
https://github.com/search?q=repo%3Aofficial-stockfish%2FStockfish%20is%3Apr%20is%3Amerged%20base%3Acluster&type=pullrequests

**Comment by BryceIsaacTiglon** [comment]
Sorry, I'm a bit unfamiliar with how everything is organized. I see the cluster branch for the current SF (but not any other versions), as well as tags to previous models. How do I access source code for say cluster version of SF 14. 

i.e. how do I access a corresponding cluster branch to a tag for an earlier SF?

**Reply by dubslow** [reply]
he said "correspond to merge commits", so you have to search in the cluster branch history for those merge commits. the most recent link he gave shows exactly what to search for, and gives very nice results of what you're looking for.
